packages/web-client/app/utils/wc-provider.ts

```diff
@@ -0,0 +1,526 @@
+import WalletConnect from '@walletconnect/client';
+import QRCodeModal from '@walletconnect/qrcode-modal';
+import HttpConnection from '@walletconnect/http-connection';
+import {
+  payloadId,
+  signingMethods,
+  parsePersonalSign,
+  getRpcUrl,
+} from '@walletconnect/utils';
+import {
+  IRPCMap,
+  IConnector,
+  IJsonRpcRequest,
+  IJsonRpcResponseSuccess,
+  IWalletConnectProviderOptions,
+  IQRCodeModalOptions,
+} from '@walletconnect/types';
+
+import ProviderEngine from 'web3-provider-engine';
+import CacheSubprovider from 'web3-provider-engine/subproviders/cache';
+import FixtureSubprovider from 'web3-provider-engine/subproviders/fixture';
+import FilterSubprovider from 'web3-provider-engine/subproviders/filters';
+import HookedWalletSubprovider from 'web3-provider-engine/subproviders/hooked-wallet';
+import NonceSubprovider from 'web3-provider-engine/subproviders/nonce-tracker';
+import SubscriptionsSubprovider from 'web3-provider-engine/subproviders/subscriptions';
+import BlockTracker from './block-tracker';
+import { WebsocketProvider } from 'web3-core';
+import { JsonRpcRequest, JsonRpcResponse } from 'json-rpc-engine';
+import { Provider } from 'eth-block-tracker';
+import { JsonRpcPayload } from 'web3-core-helpers';
+export interface ICardstackWalletConnectProviderOptions
+  extends IWalletConnectProviderOptions {
+  rpcWss: IRPCMap;
+}
+
+class WalletConnectProvider extends ProviderEngine {
+  public bridge = 'https://bridge.walletconnect.org';
+  public qrcode = true;
+  public qrcodeModal = QRCodeModal;
+  public qrcodeModalOptions: IQRCodeModalOptions | undefined = undefined;
+  public rpc: IRPCMap | null = null;
+  public rpcWss: IRPCMap;
+  public http: HttpConnection | null = null;
+  public wc: IConnector;
+  public isConnecting = false;
+  public connected = false;
+  public connectCallbacks: any[] = [];
+  public accounts: string[] = [];
+  public chainId!: number;
+  public rpcUrl = '';
+  public websocketProvider!: WebsocketProvider;
+  public networkId!: number;
+  public infuraId?: string;
+
+  constructor(opts: ICardstackWalletConnectProviderOptions) {
+    super({
+      blockTracker: new BlockTracker({
+        provider: (new WebsocketProvider(
+          opts.rpcWss[opts.chainId!]
+        ) as unknown) as Provider,
```

Comment:
This WebsocketProvider doesn't have the reassigned sendAsync from the one below and will not work, even if the import is fixed.